app/controllers/unclaimed_controller.rb

```diff
@@ -34,4 +34,21 @@ def preload_projects_from_positions
   def find_emails(email_ids)
     @emails_map = EmailAddress.where(id: email_ids.flatten).index_by(&:id)
   end
+
+  def unclaimed_people(query, find_by, per_page = nil)
+    if query && Person.find_by_name_or_email(q: query).size > OBJECT_MEMORY_CAP
+      unclaimed_people_with_limit(query, find_by)
+    else
+      Person.find_unclaimed(q: query, find_by: find_by, per_page: per_page)
+    end
+  end
+
+  # NOTE: Since this approach avoids the *includes*, it takes 3x DB time. However this prevents memory hog.
```

Comment:
And we really only expect this to be invoke when the anonymous user account is included in the query results. Is that correct?